### PR TITLE
feat(discover) Switch trace and span id to use columns instead of context

### PR DESCRIPTION
**Pull request by evanh**
Have the trace_id and span_id use the built in columns instead of the context
array.

## TODO

- [ ] add test to `tests/snuba/api/endpoints/test_organization_events_v2.py`

### CODE REVIEW

File: tests/sentry/api/test_event_search.py [hunk 1]
```diff
@@ -1079,6 +1080,14 @@ def test_transaction_status_invalid(self):
         assert "Invalid value" in six.text_type(err)
         assert "cancelled," in six.text_type(err)
 
+    def test_trace_id(self):
+        result = get_filter("trace:{}".format("a0fa8803753e40fd8124b21eeb2986b5"))
+        assert result.conditions == [["trace", "=", uuid.UUID("a0fa8803753e40fd8124b21eeb2986b5")]]
+
+    def test_trace_span_id(self):
+        result = get_filter("trace.span:{}".format("8c931f4740435fb8"))
+        assert result.conditions == [["trace.span", "=", 10129474377767673784]]
```
Comment (dashed) [inline tests/sentry/api/test_event_search.py]:
```suggestion
        assert result.conditions == [["trace.span", "=", int("8c931f4740435fb8", 16)]]
```

Comment (evanh) [reply tests/sentry/api/test_event_search.py]: It turns out that the span_id is already a string? At least, when I do this conversion it causes a type mismatch error in the tests.

File: src/sentry/api/event_search.py [hunk 2]
```diff
@@ -584,6 +585,10 @@ def convert_search_filter_to_snuba_query(search_filter):
                 )
             )
         return [name, search_filter.operator, internal_value]
+    elif name == "trace":
+        if search_filter.value.raw_value:
```
Comment (dashed) [inline src/sentry/api/event_search.py]: We can assume that `SearchValue()` is always applied to the `value` field for the `SearchFilter` namedtuple; thus,  we can assume that `search_filter.value.raw_value` always exists.

**Comment by dashed** [comment]
@evanh mind if we take over this PR?

File: src/sentry/api/event_search.py (hunk 2 above)
Comment (markstory) [reply src/sentry/api/event_search.py]: The value could be blank though in the case of `trace: foo:bar`

File: src/sentry/api/event_search.py [hunk 3]
```diff
@@ -584,6 +585,10 @@ def convert_search_filter_to_snuba_query(search_filter):
...
+            return [name, search_filter.operator, uuid.UUID(search_filter.value.raw_value)]
+        return [name, search_filter.operator, search_filter.value.raw_value]
```
Comment (markstory) [inline src/sentry/api/event_search.py]: We should raise an InvalidSearchQuery here. `trace_id` is not nullable in the transactions schema so comparing to `''` is going to fail.

File: src/sentry/api/event_search.py (hunk 2 above)
Comment (dashed) [reply src/sentry/api/event_search.py]: Oh interesting.

**Comment by dashed** [comment]
@evanh I'm going to give this back to you.

File: src/sentry/api/event_search.py [hunk 4]
```diff
@@ -584,6 +585,17 @@ def convert_search_filter_to_snuba_query(search_filter):
...
+    elif name == "trace":
+        if not search_filter.value.raw_value:
+            operator = "IS NULL" if search_filter.operator == "=" else "IS NOT NULL"
+            return [name, operator, None]
+
+        try:
+            return [name, search_filter.operator, uuid.UUID(search_filter.value.raw_value)]
+        except Exception:
+            raise InvalidSearchQuery(
+                "Invalid value for the trace condition. Accepted values are a hex string."
```
Comment (markstory) [inline src/sentry/api/event_search.py]:
```suggestion
                "Invalid value for the trace condition. Value must be are a hexadecimal UUID string."
```

**Review by markstory — APPROVED** [review]

**Pull request merged**